### PR TITLE
opt_plugin: add result uploader plugin v2

### PR DESCRIPTION
**Pull request by dmonakhov**
It would be nice to have convinient way to upload results to centralized server.
This plugin superb resultsdb which support external log URL.
Proposed implementation is very simple, but it does it's job.

Changes since V1: 
- Fixup documentation, cleanup typos
- add plugin config example
- fix up code documentation
Signed-off-by: Dmitry Monakhov <dmonakhov@openvz.org>

### CODE REVIEW

File: docs/source/optional_plugins/results.rst [hunk 1]
```diff
@@ -30,6 +30,50 @@ specify a custom location via --html . Last but not least
 the job finishes.
 
 
+Results upload Plugin
```
Comment (apahim) [inline docs/source/optional_plugins/results.rst]: Results **Upload** Plugin

File: docs/source/optional_plugins/results.rst [hunk 2]
```diff
@@ -30,6 +30,50 @@ specify a custom location via --html . Last but not least
...
+=====================
+
+This optional plugin is intended to upload the Avocado Job results to
+a dedicated sever. 
+
+To install the ResultUpload plugin from pip, use::
```
Comment (apahim) [inline docs/source/optional_plugins/results.rst]: the **Result Upload** plugin

File: docs/source/optional_plugins/results.rst [hunk 3]
```diff
@@ -30,6 +30,50 @@ specify a custom location via --html . Last but not least
...
+
+    pip install avocado-framework-plugin-result-upload
+
+Usage::
+
+    avocado run passtest.py --result-upload-url www@avocadologs.example.com:/var/www/html
+
+Avocado logs will beavailable at following URL:
```
Comment (apahim) [inline docs/source/optional_plugins/results.rst]: **be available**

File: docs/source/optional_plugins/results.rst [hunk 4]
```diff
@@ -30,6 +30,50 @@ specify a custom location via --html . Last but not least
...
+
+- ssh ::
+
+    www@avocadologs.example.com:/var/www/html/job-2017-04-21T12.54-1cefe11
+
+- html (If web server is enabled) ::
+
+    http://avocadologs.example.com/job-2017-04-21T12.54-1cefe11/
+
+Such links may be refered by other plugins, such as ResultsDB
```
Comment (apahim) [inline docs/source/optional_plugins/results.rst]: ...such as **the** ResultsDB **plugin.**

File: docs/source/optional_plugins/results.rst [hunk 5]
```diff
@@ -30,6 +30,50 @@ specify a custom location via --html . Last but not least
...
+
+    avocado run passtest.py --result-upload-url='gs://avocadolog' --result-upload-cmd='gsutil -m cp -r'
+
+You can also set the ResultUpload URL and command using a config file::
+
+    [plugins.result_upload]
+    url = www@avocadologs.example.com:/var/www/htmlavocado/job-results
+    command='rsync -arzq'
+
+And then run the Avocado command without the explicit cmd options.Notice
```
Comment (apahim) [inline docs/source/optional_plugins/results.rst]: ...**options. Notice**... (whitespace after the dot)

**Comment by dmonakhov** [comment]
Crap... I've fixed rsync in my patch, but forget to update commit. Sorry. I'll send you new PR.

**Pull request closed without merging**